src/pybind/mgr/cephadm/services/nvmeof.py

```diff
@@ -192,10 +192,15 @@ def ok_to_stop(self,
             return HandleCommandResult(-errno.EBUSY, '', warn_message)
 
         # if reached here, there is > 1 nvmeof daemon. make sure none are down
-        warn_message = ('ALERT: 1 nvmeof daemon is already down. Please bring it back up before stopping this one')
-        nvmeof_daemons = self.mgr.cache.get_daemons_by_type(self.TYPE)
-        for i in nvmeof_daemons:
-            if i.status != DaemonDescriptionStatus.running:
+        if not force:
+            warn_message = ('ALERT: Only 1 nvmeof daemon is up. Please bring 1 nvmeof daemon back up before stopping this one')
+            up_daemon_cnt = 0
+            hosts_in_maintainance_mode = self.mgr.inventory.get_host_with_state("maintenance")
```

Comment:
I think for this we should actually use `self.mgr.cache.get_unreachable_hosts()`. That grabs both the maintenance mode and offline hosts. Offline hosts have the same issue that the maintenance ones do for this case (that we can't check the daemon status there so the check that the daemon is in "running" state is not reliable) so I'd include them as well.